arch/arm/boot/dts/qcom/qcom-msm8974pro-oneplus-bacon.dts

```diff
@@ -180,6 +180,16 @@
 		interrupts-extended = <&tlmm 31 IRQ_TYPE_EDGE_FALLING>;
 		omit-battery-class;
 	};
+
+        nfc@28 {
+                compatible = "nxp,nxp-nci-i2c";
```

Comment:
You have any clue what actual NFC chipset is in the phone? Some number like NQ310 or PN547?

Reply:
Keep it like this then. Maybe we can find an ID register on the chip that give us some better idea but it's not really important